### PR TITLE
Dataset subset assignments in doc

### PR DESCRIPTION
**Pull request by matzegoebel**
Added explanation of new feature (PR #5045) that allows assigning values to a subset of a dataset to the documentation.

### CODE REVIEW

File: doc/user-guide/indexing.rst
Comment (max-sixty) [inline doc/user-guide/indexing.rst:542]: If the example is small enough such that it's possible to see the effect of the assignment, could we show the result?

**Review by max-sixty — COMMENTED** [review]
Looks good! Thanks @matzegoebel

File: doc/user-guide/indexing.rst
Comment (shoyer) [reply doc/user-guide/indexing.rst:542]: +1 it would be great to see an example that is small enough that we could show the result :)

File: doc/user-guide/indexing.rst
Comment (max-sixty) [inline doc/user-guide/indexing.rst:553]:
```suggestion
    ds
    # by integer
    ds[dict(longitude=10, latitude=10)]["u"]
    ds[dict(longitude=10, latitude=10)] = 1
    ds[dict(longitude=10, latitude=10)]["u"]
    # by label
     ds.loc[dict(longitude=47.25, level=200, latitude=[11.25, 12])]["u"]
    ds.loc[dict(longitude=47.25, level=200, latitude=[11.25, 12])] = 100
    ds.loc[dict(longitude=47.25, level=200, latitude=[11.25, 12])]["u"]
    # dataset as new values
    ds.loc[dict(longitude=48, level=200, latitude=[11.25, 12])]["u"]
    ds.loc[dict(longitude=47.25, level=200, latitude=[11.25, 12])] = ds.loc[
        dict(longitude=48.0, level=200, latitude=[11.25, 12])
    ]
    ds.loc[dict(longitude=47.25, level=200, latitude=[11.25, 12])]["u"]
```

File: doc/user-guide/indexing.rst
Comment (keewis) [inline doc/user-guide/indexing.rst:540]: can we construct the dataset manually? That way, the effect is easier to see:
```suggestion
    ds = xr.Dataset(
        {"u": (("latitude", "longitude", "level"), np.zeros(shape=(5, 2, 2)))},
        coords={
            "latitude": [10, 10.5, 11, 11.5, 12],
            "longitude": [10, 20],
            "level": [100, 200],
        },
    )
```
of course, that means we would have to update the labels below, and we might need one more variable with a `Dataset` on the right-hand side.

File: doc/user-guide/indexing.rst
Comment (keewis) [inline doc/user-guide/indexing.rst:554]: that's not a `Dataset`, is it? I think `["u"]` returns a `DataArray`?

File: doc/user-guide/indexing.rst
Comment (keewis) [inline doc/user-guide/indexing.rst:536]:
```suggestion
You can also assign values in all variables of a :py:class:`Dataset` at once:
```

File: doc/user-guide/indexing.rst
Comment (keewis) [inline doc/user-guide/indexing.rst:560]:
```suggestion
The new values must be either a scalar, a :py:class:`DataArray` or a :py:class:`Dataset` itself that contains all variables that also appear in the dataset to be modified.
```

**Review by keewis — COMMENTED** [review]
I've got a few more changes which I think might be worth changing before merging. Looks good otherwise, though.

File: doc/user-guide/indexing.rst
Comment (max-sixty) [reply doc/user-guide/indexing.rst:540]: Interesting — I rather liked loading the tutorial dataset rather than creating one specifically, since it's terser and to the extent many examples start doing this, they may become familiar.

@keewis would printing `ds` satisfy your concern? This isn't a strongly held view of mine, and I'm likely only seeing part of the context, so happy to be wrong here.

Comment (keewis) [reply doc/user-guide/indexing.rst:540]: I agree that the tutorial dataset is nice and it's better to use real-world examples rather than synthetic ones, but the effect of the operation has to be clearly visible. Also, for me an operation is always easier to understand if I can see all of the data: only with that I can try to figure out what it did, and if I can't see all the data I won't be able to verify that it didn't do anything else. This is not possible with the entire tutorial dataset (hence the suggestion), but I guess we could change this to
```suggestion
    ds = (
        xr.tutorial.open_dataset("eraint_uvz")
        .isel(latitude=slice(0, 5), longitude=slice(0, 5), level=200)
    )
```
such that we're always working with small enough data that it can be printed. I also like the use of `zeros` (we could use `xr.zeros_like` for this) because then the contrast between old and new is even higher, but if we're assigning constants that's not much of an issue.

Comment (max-sixty) [reply doc/user-guide/indexing.rst:540]: I think that's a great synthesis of the two options!

File: doc/user-guide/indexing.rst
Comment (matzegoebel) [reply doc/user-guide/indexing.rst:554]: yes right. I actually wanted to take the full dataset not only u. I changed it.

File: doc/user-guide/indexing.rst
Comment (matzegoebel) [reply doc/user-guide/indexing.rst]: I agree! In the new version I select a slice of the tutorial dataset and fill it with zeros.

**Review by keewis — COMMENTED** [review]
I've got a few more comments, but other than that we can probably merge (unless someone else has more comments?)

File: doc/user-guide/indexing.rst
Comment (matzegoebel) [reply doc/user-guide/indexing.rst:549]: good point. But maybe it's enough for this first example, so that the whole thing doesn't get too lengthy. I added ds["v"] here.

**Review by keewis — APPROVED** [review]
great, thanks for the updates. I'll merge this in a day or two if there are no further comments.

**Comment by max-sixty** [comment]
Thanks @matzegoebel ! If you'd like to add a whatsnew, please feel free to in another PR

**Comment by keewis** [comment]
> Thanks @matzegoebel ! If you'd like to add a whatsnew, please feel free to in another PR

there's already one for the new feature so the most that should be done is adding this PR to that entry (definitely optional, though).

Thanks again, @matzegoebel.

**Pull request merged**
